quickjs.c

```diff
@@ -4994,6 +5063,30 @@ static force_inline JSShapeProperty *find_own_property(JSProperty **ppr,
     return NULL;
 }
 
+static force_inline JSShapeProperty* find_own_property_ic(JSProperty** ppr, JSObject* p,
+                                                          JSAtom atom, uint32_t* offset)
+{
+    JSShape* sh;
+    JSShapeProperty *pr, *prop;
+    intptr_t h;
+    sh = p->shape;
+    h = (uintptr_t)atom & sh->prop_hash_mask;
+    h = prop_hash_end(sh)[-h - 1];
+    prop = get_shape_prop(sh);
+    while (h) {
+        pr = &prop[h - 1];
+        if (likely(pr->atom == atom)) {
+            *ppr = &p->prop[h - 1];
+            *offset = h - 1;
```

Comment:
Legibility/DRY:
```suggestion
        i = h - 1;
        pr = &prop[i];
        if (likely(pr->atom == atom)) {
            *ppr = &p->prop[i];
            *offset = i;
```
(and define `i` at the top, of course)